synapse/rest/client/v1/pusher.py

```diff
@@ -27,14 +27,47 @@
 
 
 class PusherRestServlet(ClientV1RestServlet):
-    PATTERNS = client_path_patterns("/pushers/set$")
+    PATTERNS = client_path_patterns("/pushers(/set)?$")
```

Comment:
Would probably be clearer if this was split into two, since `/pushers` is only valid for `GET` and `/pushers/set` is only valid for `PUT`